src/App.js

```diff
@@ -14,11 +14,88 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selected: [],
+  }
+
+  toAdd = (good) => {
+    if (!this.state.selected.includes(good)) {
+      this.setState(prevState => ({
+        selected: [...prevState.selected, good],
+      }));
+    }
+  }
+
+  toRemove = (good) => {
+    this.setState((prevState) => {
+      const i = prevState.selected.indexOf(good);
+
+      if (i !== -1) {
+        prevState.selected.splice(i, 1);
+      }
+
+      return {
+        selected: prevState.selected,
+      };
+    });
+  }
+
+  clearSelection = () => {
+    this.setState({
+      selected: '',
```

Comment:
'selected' in your state is initially an array. Why it becomes a string here?
That's why we need propTypes (or typeScript). We always should stick to one data type